ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/MessageDrivenComponentDescriptionFactory.java

```diff
@@ -181,22 +172,35 @@ private String getMessageListenerInterface(final CompositeIndex compositeIndex,
         return interfaces.iterator().next().toString();
     }
 
-    private Properties getActivationConfigProperties(final ActivationConfigMetaData activationConfig) {
+    private Properties getActivationConfigProperties(final MessageDrivenBeanMetaData mdb) {
         final Properties activationConfigProps = new Properties();
-        if (activationConfig == null || activationConfig.getActivationConfigProperties() == null) {
-            return activationConfigProps;
-        }
-        final ActivationConfigPropertiesMetaData activationConfigPropertiesMetaData = activationConfig.getActivationConfigProperties();
-        for (ActivationConfigPropertyMetaData activationConfigProp : activationConfigPropertiesMetaData) {
-            if (activationConfigProp == null) {
-                continue;
-            }
-            final String propName = activationConfigProp.getActivationConfigPropertyName();
-            final String propValue = activationConfigProp.getValue();
-            if (propName != null) {
-                activationConfigProps.put(propName, propValue);
+        final ActivationConfigMetaData activationConfig = mdb.getActivationConfig();
+        if (activationConfig != null && activationConfig.getActivationConfigProperties() != null) {
+            final ActivationConfigPropertiesMetaData activationConfigPropertiesMetaData = activationConfig.getActivationConfigProperties();
+            for (ActivationConfigPropertyMetaData activationConfigProp : activationConfigPropertiesMetaData) {
+                if (activationConfigProp == null) {
+                    continue;
+                }
+                final String propName = activationConfigProp.getActivationConfigPropertyName();
+                final String propValue = activationConfigProp.getValue();
+                if (propName != null) {
+                    activationConfigProps.put(propName, propValue);
+                }
             }
         }
+        if (mdb.getAcknowledgeMode() != null && !mdb.getAcknowledgeMode().isBlank()) {
+            activationConfigProps.put("acknowledgeMode", mdb.getAcknowledgeMode());
+        }
+        if (mdb.getMessageDestinationType() != null && !mdb.getMessageDestinationType().isBlank()) {
+            activationConfigProps.put("destinationType", mdb.getMessageDestinationType());
+            activationConfigProps.put("messageDestinationType", mdb.getMessageDestinationType());
```

Comment:
There are test failures caused by the above line:
```
java.lang.AssertionError: Unknown activation config property: messageDestinationType
  at org.junit.Assert.fail(Assert.java:89)
  at org.jboss.as.test.integration.ejb.management.deployments.EjbJarRuntimeResourceTestBase.testComponent(EjbJarRuntimeResourceTestBase.java:289)
  at org.jboss.as.test.integration.ejb.management.deployments.EjbJarRuntimeResourceTestBase.testNoTimerMDB(EjbJarRuntimeResourceTestBase.java:135)
```
Because `messageDestinationType` is not recognized as a valid activation config property name by the test. 

@ehsavoie Do we need this property `messageDestinationType` in addition to the similar standard one `destinationType`?

Reply:
well I wasn't sure if some resrouce adapter might be expecting it. I can simply remove it